policy/subpolicy/daily_schedule_subpolicy.go

```diff
@@ -0,0 +1,105 @@
+package subpolicy
+
+import (
+	"fmt"
+
+	"github.com/datagovsg/nomad-parametric-autoscaler/resources"
+	"github.com/datagovsg/nomad-parametric-autoscaler/utils"
+	"github.com/mitchellh/mapstructure"
+)
+
+// DailyScheduleSubPolicy is a subpolicy that extends the `SubPolicy` interface
+// and takes in the GenericSubpolicy struct
+type DailyScheduleSubPolicy struct {
+	Name             string
+	managedResources []resources.Resource
+	metadata         DailyScheduleSubPolicyMetadata
+}
+
+type ScalingWindow struct {
+	Begin int `json:"Begin"`
+	End   int `json:"End"`
+	Count int `json:"Count"`
+}
+
+func (sw ScalingWindow) IsInWindow(time int) bool {
+	return sw.Begin <= time && time < sw.End
+}
+
+// DailyScheduleSubPolicyMetadata represents metadata unique to DailyScheduleSubPolicy
+// define windows of scaling through the day and a fallback default
+type DailyScheduleSubPolicyMetadata struct {
+	Default  *int            `json:"Default"`
+	Schedule []ScalingWindow `json:"Schedule"`
+}
+
+func NewDailyScheduleSubPolicy(name string, mr []resources.Resource, meta interface{}) (*DailyScheduleSubPolicy, error) {
+	var decoded DailyScheduleSubPolicyMetadata
+	mapstructure.Decode(meta, &decoded)
+
+	if err := verifyDailyScheduleMetadata(decoded); err != nil {
+		return nil, err
+	}
+
+	return &DailyScheduleSubPolicy{
+		Name:             name,
+		managedResources: mr,
+		metadata:         decoded,
+	}, nil
+}
+
+func verifyDailyScheduleMetadata(meta DailyScheduleSubPolicyMetadata) error {
```

Comment:
too bad `mapstructure` doesn't support the struct tag `binding:"Required"` like in Gin, otherwise this wouldnt be necessary

https://github.com/golang/go/wiki/Well-known-struct-tags